util/errors/errors.go

```diff
@@ -12,15 +12,12 @@ import (
 
 	log "github.com/sirupsen/logrus"
 	apierr "k8s.io/apimachinery/pkg/api/errors"
-
-	argoerrs "github.com/argoproj/argo-workflows/v3/errors"
 )
 
 func IsTransientErr(err error) bool {
 	if err == nil {
 		return false
 	}
-	err = argoerrs.Cause(err)
```

Comment:
I think we need to be careful here, we should probably be unwrapping errors. Funcs like `isExcheedeQuotaErr` maybe need to be updated to use `errors.Is()`